service/worker/archiver/activities.go

```diff
@@ -209,28 +250,39 @@ func uploadHistoryActivity(ctx context.Context, request ArchiveRequest) (err err
 		}
 		if err := uploadBlob(ctx, blobstoreClient, request.BucketName, key, blob); err != nil {
 			logger.Error(uploadErrorMsg, tag.ArchivalUploadFailReason(errorDetails(err)), tag.ArchivalBlobKey(key.String()), tag.Error(err))
-			return err
+			return result, err
 		}
 		handledLastBlob = *historyBlob.Header.IsLast
+
+		progress.UploadedBlobs = append(progress.UploadedBlobs, key.String())
+		currIteratorState, err := historyBlobIterator.GetState()
+		if err == nil {
+			progress.IteratorState = currIteratorState
+			progress.BlobPageToken = pageToken
```

Comment:
It feels weird to me that BlobPageToken and UploadedBlobs can become out of sync. The way I am reading this code is that progress.BlobPageToken represents the highest blob page you have successfully uploaded so far. But in the case where historyBlobIterator.GetState() returns error you get a progress which has a blob in UploadedBlobs with a higher page than progress.BlobPageToken

Reply:
Updated, but I still feel a little bit weird as I think progress.UploadedBlobs should always reflect what has been uploaded. In the case when GetState() returns an error, the blob is still uploaded successfully, so I think progress should still be updated.